src/Calendar.php

```diff
@@ -462,4 +464,18 @@ private function deriveBookingAvailabilityFromSubEvents(): BookingAvailability
 
         return BookingAvailability::unavailable();
     }
+
+    /**
+     * A projection can require a potential fix:
+     * - For a periodic or permanent calendar this is always available
+     * - If there are timestamps the top level status is calculated
+     */
+    private function determineCorrectTopBookingAvailabilityForProjection(): BookingAvailability
+    {
+        if (!$this->allowsUpdatingBookingAvailability()) {
```

Comment:
Again, here I would just check if the event has subEvents or not. You can only calculate the top availability if there are subEvents anyway, regardless of which calendar types can be updated. If there are no subEvents, there are no booking availabilities to use to calculate it anyway.